### ISSUE TITLE
Turn `+data-root+` into a `data-path`

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#1846]
Current `+data-root+` is 

```lisp
(defvar +data-root+ "nyxt")
```

And `expand-default-path` expands to `(uiop:xdg-data-home +data-root+)` by default.

A problem arises if we try to move all "data-home" files to somewhere else, as is the case with the `dev` profile documented in the manual:

```lisp
(define-class dev-data-profile (data-profile)
   ((name :initform \"dev\"))
   (:documentation \"Development profile.\"))
(defmethod nyxt:expand-data-path ((profile dev-data-profile) (path data-path))
  \"Persist data to /tmp/nyxt/.\"
  (expand-default-path (make-instance (class-name (class-of path))
                                      :basename (basename path)
                                      :dirname \"/tmp/nyxt/\")))
```

Here, any data-path value that specifies a _directory_ gets overridden and stored in `/tmp/nyxt/$BASENAME` directly.  This is unfortunate.

So instead I propose to turn `+data-root+` into a `data-path` itself, so that it would be easy to define a profile and specialize the `expand-path` on `+data-root+` to expand to somewhere new.  Thus the `dev` profile definition could be as simple as

```lisp
(defmethod nyxt:expand-data-path ((profile dev-data-profile) (+data-root+ (eql '+data-root+)))
  "Persist data to /tmp/nyxt/."
  "/tmp/nyxt/")
```

Thoughts?

**Comment by Ambrevar** [comment]
We could fix this together with #1270.

**Comment by jmercouris** [comment]
I agree with this change. However, I disagree with many of the challenges of the existing data path API. That would be a discussion for a different issue!
